stable/database/files/readinessprobe

```diff
@@ -38,6 +38,9 @@ parse_fields() {
 pgrep -xa nuodb | (
     n=0
     while read pid argv; do
+        # skip zombie processes
+        [ "$(ps -o stat= -p "$pid")" != Z ] || continue
```

Comment:
Even though we don't invoke `set -e`, a lot of our scripts do, so I tend to avoid commands that return error exit codes.

What would happen if we do the following? 
```
set -e
...

[ "$(ps -o stat= -p "$pid")" = Z ] && continue
```

Reply:
First, IMO you [shouldn't use](https://mywiki.wooledge.org/BashFAQ/105) "set -e".  It's a terribly misdesigned "feature" with many bugs and inconsistencies between different shell implementations.  But anyway, it works "the right way"

```
~$ /bin/sh -ec '[ foo = bar ]; echo there'
~$
```
shows that it exits immediately for "normal", but:
```
~$ /bin/sh -ec '[ foo = bar ] && echo hi; echo there'
there
~$
```
shows it doesn't exit and still runs the commands if you add a `&&`

Reply:
Okay, I resolved this by using an `if`-statement.